### PR TITLE
Device lookup page - support additional search fields

### LINKED ISSUE #4438: Device code lookup: Display and allow user to search by additional fields for devices
Replace existing device dropdown/combo box with a view that shows:
- manufacturer
- equipment model name
- nickname 

Allow user to search by any of those values.

Figma: https://www.figma.com/file/NPSPM5SBzQ7Ra3mmr9M9cB/SimpleReport-Public-Site?node-id=13176%3A34613

Design:
- figure out label for nickname

### PR DESCRIPTION
**Pull request by nathancrtr**
## Related Issue
- Closes #4438 

## Changes Proposed
- The device lookup feature searches multiple fields:
    - `name`
    - `model`
    - `manufacturer`
- 

## Additional Information
- decisions that were made
- notice of future work that needs to be done

## Screenshots / Demos

Screen Shot 2022-10-26 at 2 38 41 PM
Screen Shot 2022-10-26 at 2 38 51 PM
Screen Shot 2022-11-02 at 1 42 13 PM

Screen Shot 2022-11-02 at 1 41 54 PM

Screen Shot 2022-10-26 at 2 41 50 PM

## Feedback requested
- Following the pattern set by the previous iteration of this view, the full device list is retrieved by the container component **once**
- User searches filter through the in-memory device list
- @mehansen @DanielSass any concerns about this? We have a hundred-something devices in prod, which feels like a small enough result set that in-memory array operations should be pretty zippy, but we can hit the network for each search if anyone has concerns
- @vz3 @lauraykerr @lg-king some design and usability questions that arose during this work:
    - Do the "Test result" values vary between devices? If not, is still still valuable to include?
        - Screen Shot 2022-10-26 at 2 49 00 PM
    - Please review the main header
        - The design specifies the title as `Lookup device codes`, but IME "lookup" is only used as a noun with "look up" being the verb form
    - The previous iteration of this feature used a combobox for the search input field, which caused a dropdown to appear even before any query has been entered
        - Screen Shot 2022-10-26 at 2 54 39 PM
        - Do you think users would still find this helpful? I could imagine a user being unsure about what they should enter into the search field

### CODE REVIEW

**Review by DanielSass — DISMISSED** [review]
awesome work, lgtm

**Comment by mehansen** [comment]
From the Figma it looks like the test type column is meant to refer to what disease the device supports. did that get changed?

**Comment by mehansen** [comment]
> @mehansen @DanielSass any concerns about this? We have a hundred-something devices in prod, which feels like a small enough result set that in-memory array operations should be pretty zippy, but we can hit the network for each search if anyone has concerns

I think that makes sense!

>The design specifies the title as Lookup device codes, but IME "lookup" is only used as a noun with "look up" being the verb form

nice callout this was bugging me too but I couldn't put my finger on why it seemed weird. if we're trying to follow a pattern of starting page titles with verbs, maybe "retrieve device codes" or just "get device codes" could work
@lauraykerr

**Comment by lauraykerr** [comment]
@nathancrtr Great catch on the title. Our most common pattern is using a noun phrase for the page title ("Upload your results" on the uploader page is an exception), so would recommend `Device code lookup`. 

For the question about the test code, I believe the code doesn't vary between devices but it's still useful to include here since the copyable field is convenient and many of the other "order and result" fields appear here, but curious what @vz3 thinks.

**Comment by nathancrtr** [comment]
> @nathancrtr Great catch on the title. Our most common pattern is using a noun phrase for the page title ("Upload your results" on the uploader page is an exception), so would recommend `Device code lookup`.
> 
> For the question about the test code, I believe the code doesn't vary between devices but it's still useful to include here since the copyable field is convenient and many of the other "order and result" fields appear here, but curious what @vz3 thinks.

Sounds good to me @lauraykerr ! Changed the page title & added the test result codes in [7b6b9c1](https://github.com/CDCgov/prime-simplereport/pull/4593/commits/7b6b9c1c7ae9468b48b7db22e38060dc09c1366d).

File: frontend/src/app/uploads/DeviceLookup/DeviceSearchResults.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    resultsContent = <p>Searching...</p>;
+  } else if (devices.length === 0) {
+    resultsContent = (
+      <div
+        className={
+          "display-flex flex-column flex-align-center margin-x-7 margin-y-2"
+        }
+      >
+        <div className="margin-bottom-105">
+          No device found matching <strong>{queryString}</strong>
```
Comment (mehansen) [inline frontend/src/app/uploads/DeviceLookup/DeviceSearchResults.tsx:39]: copy nit: period after this? we have it in our other 'no match' copy when searching for patients in the queue page or the results page

**Review by mehansen — APPROVED** [review]
lgtm! got some random nits/questions but nothing blocking!
breaking out DeviceDetails was nice 👌

File: frontend/src/app/uploads/DeviceLookup/DeviceLookup.tsx [hunk 2]
```diff
@@ -1,211 +1,124 @@
...
-            <FontAwesomeIcon
-              icon={copiedSlug === copiedAttribute ? faCheck : faCopy}
-            />
-          </button>
-        )}
-        {device && copiedSlug === copiedAttribute && <CopyTooltip />}
-      </div>
-    );
+  const onSearchClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
```
[the remark this answers is not in the document]
Comment (nathancrtr) [reply frontend/src/app/uploads/DeviceLookup/DeviceLookup.tsx:83]: You see this attached to submission handlers pretty often . The "default" behavior when submitting a form is for the browser to refresh the page. We want to suppress that here.

[the remark this answers is not in the document]
Comment (mehansen) [reply frontend/src/app/uploads/DeviceLookup/DeviceLookup.tsx:83]: gotcha - thanks!

**Comment by mehansen** [comment]
I think you might need to merge `main` after the node upgrade

**Comment by vz3** [comment]
@nathancrtr I believe there are devices that use different values for different test results but I'll dig into the data and double check

**Pull request merged**